pageserver/src/tenant/timeline.rs

```diff
@@ -3127,55 +3127,57 @@ impl Timeline {
             unmapped_keyspace.remove_overlapping_with(&keys_done_last_step);
             completed_keyspace.merge(&keys_done_last_step);
 
-            let guard = timeline.layers.read().await;
-            let layers = guard.layer_map();
+            if unmapped_keyspace.start().is_some() {
```

Comment:
1. An `is_empty()` function would be more expressive. Let's add one?
2. There should be a brief comment explaining that this was shown to be important for performance.

Reply:
Yeah either a `KeySpace::is_empty` function or `unmapped_keyspace.ranges.is_empty()` would be nicer here.

Reply:
Good point: [e57853d](https://github.com/neondatabase/neon/pull/7490/commits/e57853decba2b268cdc137278f0ef9903bee323b)